### PR TITLE
Enhancement: No results messages link to docs

### PR DESCRIPTION
**Pull request by znicholasbrown**
Note for reviewers: decided to try something slightly new here, using the markdown renderer for localization content; seems to work nicely but also open to other suggestions.

### CODE REVIEW

File: src/localization/locale/en.ts [hunk 1]
```diff
@@ -87,7 +88,7 @@ export const en = {
   },
   info: {
     artifact: 'Artifact',
-    noResults: 'No tracked results',
+    noResults: 'No tracked results, enable [result persistence](https://docs.prefect.io/concepts/results/#persisting-results) to track results.',
```
Comment (madkinsz) [inline src/localization/locale/en.ts:91]: This only affects old client versions basically right?

Comment (madkinsz) [reply src/localization/locale/en.ts:91]: Won't they need to upgrade to a version with result artifacts? Maybe we should just suggest being on the latest version.

Comment (znicholasbrown) [reply src/localization/locale/en.ts:91]: Afaik, yeah. It'll show up for empty states for task runs or the flow run itself as well which at the very least is a helpful "what this feature is"

**Comment by pleek91** [comment]
@znicholasbrown does the markdown rendered look the same as if we did this with html like we normally would? I'm thinking the link. Does that use `p-link` and get the external url icon? 

Otherwise I love this idea.

**Review by pleek91 — APPROVED** [review]

**Pull request merged**